src/components/TypingIndicator/TypingIndicator.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+
+import { AvatarProps, Avatar as DefaultAvatar } from '../Avatar';
+
+import { useChannelContext } from '../../context/ChannelContext';
+
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../../types/types';
+
+export type TypingIndicatorProps = {
+  /**
+   * Custom UI component to display user avatar.
+   *
+   * Defaults to and accepts same props as: [Avatar](https://github.com/GetStream/stream-chat-react/blob/master/src/components/Avatar/Avatar.tsx)
+   * */
+  Avatar?: React.ComponentType<AvatarProps>;
+  /** Size in pixels and the default is 32px */
+  avatarSize?: number;
+  /** Boolean for it there is a threadlist */
+  threadList?: boolean;
+};
+
+/**
+ * TypingIndicator lists users currently typing, it needs to be a child of Channel component
+ */
+const UnMemoizedTypingIndicator = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends DefaultChannelType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends DefaultUserType = DefaultUserType
+>(
+  props: TypingIndicatorProps,
+) => {
+  const { Avatar = DefaultAvatar, avatarSize = 32, threadList } = props;
+
+  const { channel, client, thread, typing } = useChannelContext<
```

Comment:
let's pull client off ChatContext, ideally we should always be pulling from there, might make tests fail if the only have ChannelContext.Provider wrapping the mock component